data/src/test/java/org/apache/iceberg/io/TestWriterMetrics.java

```diff
@@ -227,23 +227,23 @@ public void testPositionDeleteMetricsCoveringMultipleDataFiles() throws IOExcept
 
     // should have NO bounds for path and position as the file covers multiple data paths
     Map<Integer, ByteBuffer> lowerBounds = deleteFile.lowerBounds();
-    Assert.assertEquals(2, lowerBounds.size());
-    Assert.assertEquals(
-        3, (int) Conversions.fromByteBuffer(Types.IntegerType.get(), lowerBounds.get(1)));
-    Assert.assertEquals(
-        3L, (long) Conversions.fromByteBuffer(Types.LongType.get(), lowerBounds.get(5)));
+    assertThat(lowerBounds).hasSize(2);
+    assertThat((int) Conversions.fromByteBuffer(Types.IntegerType.get(), lowerBounds.get(1)))
+        .isEqualTo(3);
+    assertThat((long) Conversions.fromByteBuffer(Types.LongType.get(), lowerBounds.get(5)))
+        .isEqualTo(3L);
 
     Map<Integer, ByteBuffer> upperBounds = deleteFile.upperBounds();
-    Assert.assertEquals(2, upperBounds.size());
-    Assert.assertEquals(
-        3, (int) Conversions.fromByteBuffer(Types.IntegerType.get(), upperBounds.get(1)));
-    Assert.assertEquals(
-        3L, (long) Conversions.fromByteBuffer(Types.LongType.get(), upperBounds.get(5)));
+    assertThat(upperBounds).hasSize(2);
+    assertThat((int) Conversions.fromByteBuffer(Types.IntegerType.get(), upperBounds.get(1)))
+        .isEqualTo(3);
+    assertThat((long) Conversions.fromByteBuffer(Types.LongType.get(), upperBounds.get(5)))
+        .isEqualTo(3L);
   }
 
-  @Test
+  @TestTemplate
   public void testMaxColumns() throws IOException {
-    File tableDir = temp.newFolder();
+    File tableDir = new File(tempDir, "table");
```

Comment:
should use same mechanism as in other places to create a new folder. Please make sure this is properly handled in all places within this PR

Reply:
OK, it's fixed now in all this PR.